azure/converters/rules_test.go

```diff
@@ -17,47 +17,49 @@ import (
 	"testing"
 
 	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-08-01/network"
-	"github.com/stretchr/testify/assert"
+	"github.com/onsi/gomega"
 	"k8s.io/utils/pointer"
 	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 )
 
 func TestSecurityRuleToSDK(t *testing.T) {
+	g := gomega.NewWithT(t)
+
 	// Test case for a security rule with all fields provided
 	rule := infrav1.SecurityRule{
-		Name: "test-rule",
-		Description: "Test security rule",
-		Source: "10.0.0.0/24",
-		SourcePorts: "80,443",
-		Destination: "192.168.0.0/16",
+		Name:             "test-rule",
+		Description:      "Test security rule",
+		Source:           "10.0.0.0/24",
```

Comment:
The fields `Source`, `SourcePorts`, `Destination`, and `DestinationPorts` are string pointers. You will need to wrap them with `pointers.String()`

Example here: https://github.com/willie-yao/cluster-api-provider-azure/blob/azuremachine-controller-test/azure/services/securitygroups/spec_test.go#L48